### PR TITLE
feat: add support for Q&A Schema App input

### PR DESCRIPTION
**Pull request by rossiam**
Add support for Q&A input for Schema Apps.

## Checklist

- [x] I have read the **[CONTRIBUTING](../CONTRIBUTING.md)** document
- [x] Any required documentation has been added
- [ ] My code follows the code style of this project (`npm run lint` produces no warnings/errors)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Review by bflorian — CHANGES_REQUESTED** [review]
A couple of nits and a small UI suggestion

File: packages/cli/src/lib/commands/schema-util.ts [hunk 1]
```diff
@@ -1 +1,132 @@
...
+			default: (context?: unknown[]) => (context?.[0] as Pick<SchemaAppRequest, 'partnerName'>)?.partnerName ?? '',
+		}),
+		oAuthAuthorizationUrl: stringDef('OAuth Authorization URL', { validate: httpsURLValidate }),
+		icon: optionalStringDef('Icon URL', { validate: httpsURLValidate }),
+		icon2x: optionalStringDef('2x Icon URL', { validate: httpsURLValidate }),
+		icon3x: optionalStringDef('3x Icon URL', { validate: httpsURLValidate }),
+		oAuthClientId: stringDef('Partner OAuth Client Id'),
+		oAuthClientSecret: stringDef('Partner OAuth Client Secret'),
+		oAuthTokenUrl: stringDef('Partner OAuth Refresh Token URL', { validate: httpsURLValidate }),
+		oAuthScope: stringDef('Partner OAuth Scope'),
```
[the remark this answers is not in the document]
Comment (rossiam) [reply packages/cli/src/lib/commands/schema-util.ts]: Fixed.

File: packages/cli/src/lib/commands/schema-util.ts [hunk 2]
```diff
@@ -1 +1,132 @@
...
+		appName: optionalStringDef('App Name', {
+			default: (context?: unknown[]) => (context?.[0] as Pick<SchemaAppRequest, 'partnerName'>)?.partnerName ?? '',
+		}),
+		oAuthAuthorizationUrl: stringDef('OAuth Authorization URL', { validate: httpsURLValidate }),
+		icon: optionalStringDef('Icon URL', { validate: httpsURLValidate }),
+		icon2x: optionalStringDef('2x Icon URL', { validate: httpsURLValidate }),
+		icon3x: optionalStringDef('3x Icon URL', { validate: httpsURLValidate }),
+		oAuthClientId: stringDef('Partner OAuth Client Id'),
+		oAuthClientSecret: stringDef('Partner OAuth Client Secret'),
+		oAuthTokenUrl: stringDef('Partner OAuth Refresh Token URL', { validate: httpsURLValidate }),
```
[the remark this answers is not in the document]
Comment (rossiam) [reply packages/cli/src/lib/commands/schema-util.ts]: Fixed.

File: packages/cli/src/lib/commands/schema-util.ts [hunk 3]
```diff
@@ -1 +1,137 @@
...
 export const SCHEMA_AWS_PRINCIPAL = '148790070172'
+
+const arnDef = (name: string, inChina: boolean, initialValue?: SchemaAppRequest, options?: { forChina?: boolean }): InputDefinition<string | undefined> => {
+	if (inChina && !options?.forChina || !inChina && options?.forChina) {
+		return undefinedDef
+	}
+
+	const helpText = awsHelpText
+	const initiallyActive = initialValue?.hostingType === 'lambda'
+	console.error(`initiallyActive is ${initiallyActive} for ${name} because initialValue?.hostingType = ${initialValue?.hostingType}`)
```
[the remark this answers is not in the document]
Comment (rossiam) [reply packages/cli/src/lib/commands/schema-util.ts]: Oops, meant to take that out. Fixed.

File: packages/cli/src/lib/commands/schema-util.ts [hunk 4]
```diff
@@ -1 +1,137 @@
...
+		{ initiallyActive })
+}
+
+const webHookUrlDef = (inChina: boolean, initialValue?: SchemaAppRequest): InputDefinition<string | undefined> => {
+	if (inChina) {
+		return undefinedDef
+	}
+
+	const initiallyActive = initialValue?.hostingType === 'webhook'
+	console.error(`initiallyActive is ${initiallyActive} for Webhook URL because initialValue?.hostingType = ${initialValue?.hostingType}`)
```
[the remark this answers is not in the document]
Comment (rossiam) [reply packages/cli/src/lib/commands/schema-util.ts]: Oops, didn't mean to leave that in. Fixed.

**Pull request merged**